### ISSUE TITLE
Unable to retrieve last element of column header

### ISSUE THREAD

**Issue by swopnil123** [ronkok/Hurmet#37]
I'm not able to retreive the last element of column header when using Table1.A It will only work if I do Table1.A_end

```
¢` format = "h4" `

#### Vertical Distribution of Seismic Force

From earlier analysis

::: indented
¢` S_DS = 1.0 `          ¢` I = 1.0 `

¢` V_base = 400kips `, base shear        ¢` T = 0.75 sec `, fundamental period
of the structure
:::

From ASCE-16 section 12.8.3, Equivalent Lateral Procedure

::: indented
¢` k = {   1 if T ≤ 0.5;   2 if T ≥ 2.5;   1 + (T - 0.5) // 2 otherwise } = ? `

¢` C_vs = (w × h^k) / (Σ (w × h^k)) `, vertical distribution coefficient, Eq.
12.8-12

¢` F = C_vs V_base `, local force, Eq. 12.8-11
:::

¶

: dist: Vertical Distribution of Seismic Force
+--------+----------+---------+----------+-----------+--------------+--------+
| Floor  | Weight\  | Height\ | w × h^k^ | C~vs~     | Force\       | Shear\ |
|        | kips     | ft      |          |           | kips         | kips   |
+========+:========:+:=======:+:========:+:=========:+:============:+:======:+
| roof   | 950      | 70      | =B1×C1^k | =D1/D_end | ==E1× V_base | =F1    |
+--------+----------+---------+----------+-----------+--------------+--------+
| fifth  | 1,250    | 56      | "        | "         | "            | =G1+F2 |
+--------+----------+---------+----------+-----------+--------------+--------+
| fourth | "        | 42      | "        | "         | "            | "      |
+--------+----------+---------+----------+-----------+--------------+--------+
| third  | "        | 28      | "        | "         | "            | "      |
+--------+----------+---------+----------+-----------+--------------+--------+
| second | "        | 14      | "        | "         | "            | "      |
+--------+----------+---------+----------+-----------+--------------+--------+
| total  | =sum(up) | 0       | =sum(up) |           | =sum(up)     | =F_end |
+--------+----------+---------+----------+-----------+--------------+--------+
{#dist ."four-rules spreadsheet" colWidths="NaN NaN NaN NaN NaN NaN NaN"}

¶

¢` dist.A=? `, ¢` dist.A6=? `

¢` dist.B=? `
```

**Comment by swopnil123** [comment]
Also can you look at this one? Why I'm having issues with this table?

```
¢` CemPlast = 135pcf `, Density of Cement Plaster 

¢` RigdIns = 1.5'psf/in' `, Weight per inch 

¢` Gypsheath = 2.8psf `, Weight of Gypsum Sheathing 5/8” Thickness

¢` MetalStuds = 3psf `, Metal studs at 16” OC

¢` BattIns = 0.5psf `, Batt Insulation 

: SA - Metal Stud Wall Weigh up
+---------------------------+--------------+---------------+-----------------------------+
| **Item**                  | **Thickness\ | **Weight\     | **Notes**                   |
|                           | in**         | psf**         |                             |
+===========================+==============+===============+=============================+
| 7/8 Cement Plaster        | =7/8         | ==B1*CemPlast |                             |
+---------------------------+--------------+---------------+-----------------------------+
| 1 1/2” Rigid Insulation   | = 1 1/2      | ==B2*RigdIns  | 1.5psf per inch             |
+---------------------------+--------------+---------------+-----------------------------+
| (1) 5/8” Gypsum Sheathing |              | ==1*Gypsheath | 1 Layer of Gypsum Sheathing |
+---------------------------+--------------+---------------+-----------------------------+
| Metal Studs at 16” OC     |              | ==MetalStuds  | Metal Studs at 16”OC        |
+---------------------------+--------------+---------------+-----------------------------+
| (1) Batt Insulation       |              | ==BattIns     | 1 Layer of Batt Insulation  |
+---------------------------+--------------+---------------+-----------------------------+
| (1) 5/8” Gypsum Sheathing |              | ==1*Gypsheath | 1 Layer of Gypsum Sheathing |
+---------------------------+--------------+---------------+-----------------------------+
| Misc                      |              | 2             |                             |
+---------------------------+--------------+---------------+-----------------------------+
| Total                     |              | =sum(up)      |                             |
+---------------------------+--------------+---------------+-----------------------------+
{#SA ."four-rules spreadsheet" colWidths="null null 163 null"}

¢` SA.C=??psf `

¢` SA.B=??in `

```

**Comment by swopnil123** [comment]
> Also can you look at this one? Why I'm having issues with this table? It does not work with SA.B ?
> 
> ```
> ¢` CemPlast = 135pcf `, Density of Cement Plaster 
> 
> ¢` RigdIns = 1.5'psf/in' `, Weight per inch 
> 
> ¢` Gypsheath = 2.8psf `, Weight of Gypsum Sheathing 5/8” Thickness
> 
> ¢` MetalStuds = 3psf `, Metal studs at 16” OC
> 
> ¢` BattIns = 0.5psf `, Batt Insulation 
> 
> : SA - Metal Stud Wall Weigh up
> +---------------------------+--------------+---------------+-----------------------------+
> | **Item**                  | **Thickness\ | **Weight\     | **Notes**                   |
> |                           | in**         | psf**         |                             |
> +===========================+==============+===============+=============================+
> | 7/8 Cement Plaster        | =7/8         | ==B1*CemPlast |                             |
> +---------------------------+--------------+---------------+-----------------------------+
> | 1 1/2” Rigid Insulation   | = 1 1/2      | ==B2*RigdIns  | 1.5psf per inch             |
> +---------------------------+--------------+---------------+-----------------------------+
> | (1) 5/8” Gypsum Sheathing |              | ==1*Gypsheath | 1 Layer of Gypsum Sheathing |
> +---------------------------+--------------+---------------+-----------------------------+
> | Metal Studs at 16” OC     |              | ==MetalStuds  | Metal Studs at 16”OC        |
> +---------------------------+--------------+---------------+-----------------------------+
> | (1) Batt Insulation       |              | ==BattIns     | 1 Layer of Batt Insulation  |
> +---------------------------+--------------+---------------+-----------------------------+
> | (1) 5/8” Gypsum Sheathing |              | ==1*Gypsheath | 1 Layer of Gypsum Sheathing |
> +---------------------------+--------------+---------------+-----------------------------+
> | Misc                      |              | 2             |                             |
> +---------------------------+--------------+---------------+-----------------------------+
> | Total                     |              | =sum(up)      |                             |
> +---------------------------+--------------+---------------+-----------------------------+
> {#SA ."four-rules spreadsheet" colWidths="null null 163 null"}
> 
> ¢` SA.C=??psf `
> 
> ¢` SA.B=??in `
> ```

I think I got the answer for this one. It's because the rows are empty.

**Comment by ronkok** [comment]
I'll look into it.

**Comment by ronkok** [comment]
I have fixed the bug that caused the problem in your first example.

Regarding your second example, remember that in a Hurmet vector, all the elements must be of the same type. But column B in the spreadsheet contains a mixture of types. The first two elements are numbers and the rest of the column is blank. To form a valid vector, those elements could be zeros, but they cannot be blank.

I've revised the code with a better error message.

Thank you for bringing this to my attention.
